### PR TITLE
FIX: Issue 414 - failing login for domain change

### PR DESCRIPTION
**Pull request by ritwik-g**
## What

- Fix related to https://github.com/Zipstack/unstract/issues/414
- Changed logic for parsing SESSION_COOKIE_SECURE and CSRF_COOKIE_SECURE from env. Earlier one was not parsing bool properly
- Made these `False` in sample.env 
- Need to update the doc to use `True` if they are using https. Also need to make sure in backend/.env we are using https:// for urls when using HTTPS connection

## Why

- Looks like for localhost the browser is not considering the secure or non secure. But when the domain changes browser is skipping the cookie setting since we are responding with `secure` cookie but the communication channel is unsecure. So basically further calls are failing because we don't have the session cookies.

image (5)

## How

- By making this `False` by default in sample.env any one trying to use this should have no issues. 

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

-

## Database Migrations

- 

## Env Config

- 

## Relevant Docs

-

## Related Issues or PRs

-

## Dependencies Versions

-

## Notes on Testing

-

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

**Review by hari-kuriakose — APPROVED** [review]
@ritwik-g LGTM

File: backend/backend/settings/base.py [hunk 1]
```diff
@@ -86,8 +86,12 @@ def get_required_setting(
...
     "SESSION_EXPIRATION_TIME_IN_SECOND", 3600
 )
-SESSION_COOKIE_SECURE = os.environ.get("SESSION_COOKIE_SECURE", True)
-CSRF_COOKIE_SECURE = os.environ.get("CSRF_COOKIE_SECURE", True)
+SESSION_COOKIE_SECURE = CommonUtils.str_to_bool(
+    os.environ.get("SESSION_COOKIE_SECURE", "True")
+)
+CSRF_COOKIE_SECURE = CommonUtils.str_to_bool(
+    os.environ.get("CSRF_COOKIE_SECURE", "True")
+)
```
Comment (hari-kuriakose) [inline backend/backend/settings/base.py:94]: @ritwik-g Might be better to have `False` here as default to cover all use cases.

Will also update the public docs accordingly.

Comment (hari-kuriakose) [reply backend/backend/settings/base.py:94]: cc @athul-rs

Comment (hari-kuriakose) [inline backend/backend/settings/base.py:94]:
```suggestion
SESSION_COOKIE_SECURE = CommonUtils.str_to_bool(
    os.environ.get("SESSION_COOKIE_SECURE", "False")
)
CSRF_COOKIE_SECURE = CommonUtils.str_to_bool(
    os.environ.get("CSRF_COOKIE_SECURE", "False")
)
```

Comment (ritwik-g) [reply backend/backend/settings/base.py:94]:
```suggestion
SESSION_COOKIE_SECURE = CommonUtils.str_to_bool(
    os.environ.get("SESSION_COOKIE_SECURE", "False")
)
CSRF_COOKIE_SECURE = CommonUtils.str_to_bool(
    os.environ.get("CSRF_COOKIE_SECURE", "False")
)
```

File: backend/sample.env [hunk 2]
```diff
@@ -1,5 +1,9 @@
 DJANGO_SETTINGS_MODULE='backend.settings.dev'
 
+# NOTE: Change below to True or comment it out if you are running in HTTPS mode.
```
Comment (ritwik-g) [inline backend/sample.env:3]:
```suggestion
# NOTE: Change below to True if you are running in HTTPS mode.
```

**Review by muhammad-ali-e — APPROVED** [review]
LGTM

**Review by athul-rs — APPROVED** [review]

**Review by chandrasekharan-zipstack — APPROVED** [review]

**Pull request merged**